### ISSUE TITLE
deploy from python 3

### ISSUE THREAD

**Issue by lheagy** [simpeg/simpeg#560]
docs and pypi deployments should be done from the python 3 builds on travis ci. currently they are done on 2.7.

**Comment by lheagy** [comment]
this is causing pypi installation failures on python 3

**Comment by rowanc1** [comment]
Would this change anything on the 2.7 release on PyPI? Probably should be encouraging people to use Python3 anyways...

**Comment by lheagy** [comment]
(not sure... we can test on a --pre deployment)

**Comment by lheagy** [comment]
right now, the docs need to remain on python2 (we are using webapp2, which currently depends on python2). https://webapp2.readthedocs.io/en/latest/

The pypi deploy, however, should still be on python 3

**Comment by lheagy** [comment]
@rowanc1, @bsmithyman : do you have any recommendations for running webapp2 (or something similar??) with python3?

**Comment by rowanc1** [comment]
Because we are deploying the docs to appengine, that environment only has py2.7 support. Webapp2 is in pre release for python 3, but we only need that for docs - which has to be in py2.7 anyways.

**Comment by lheagy** [comment]
kk, thanks! the docs are still not deploying properly :( (They haven't since the addition of the gallery...) https://travis-ci.org/simpeg/simpeg/jobs/216589157 Any ideas? I can try hitting on this later today if nothing jumps out.

**Comment by rowanc1** [comment]
`gcloud -q preview app deploy ./docs/app.yaml --version ${TRAVIS_COMMIT} --promote;`

I think that the `preview` is no longer necessary.

I also think that we need to specify the project in there as well? (in addition to the `set project` that is already there)

Something like:

```bash
if [ "$TRAVIS_BRANCH" = "master" ]; then
  gcloud auth activate-service-account --key-file credentials/client-secret.json;
  export GAE_PROJECT=simpegdocs;
else
  gcloud auth activate-service-account --key-file credentials/client-secret-dev.json;
  export GAE_PROJECT=simpeg-dev-docs;
fi;
gcloud config set project $GAE_PROJECT;
gcloud app deploy ./docs/app.yaml --project $GAE_PROJECT --version ${TRAVIS_COMMIT} --promote;
```

Notice that I added the: `--project $GAE_PROJECT`, this was me looking at:
http://stackoverflow.com/questions/33493750/error-response-13-an-internal-error-occurred-when-deploying-project-on-googl

I am not sure that this is necessary anymore either:
`gcloud -q components update gae-python;`
travis seems to be complaining.

**Comment by lheagy** [comment]
This closed with #570. Deploying the docs from python 3 will have to wait until google app engine runs on python 3
